### PR TITLE
Final Pull Request ROR Group CapStone

### PR DESCRIPTION
**Pull request by paulshaun-kdtk**
In this project:

- [ ] We implemented functionalities to meet the [project](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md) requirements.

- we created the recipe app
- each member of the group covered a separate component of the app
- one member covered the foods (including specs and ui)
- the other covered the recipes (including specs and ui)
- the other covered inventory (including specs and ui)

### CODE REVIEW

**Review by faisalakb — APPROVED** [review]
Project is approved.

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @is_active = true
+    # @foods = current_user.foods
+    if params[:id] == 'missing_foods'
```
Comment (OyePriscilla) [inline app/controllers/foods_controller.rb:7]: - [ ] Please kindly remove this commented code in order to make your codebase cleaner. 👍

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        <div class="field">
+          <%= f.label :password, class: "label" %><br />
+          <%= f.password_field :password, autocomplete: "current-password", class: "password" %>
+        </div>
+
+        <% if devise_mapping.rememberable? %>
+          <div class="field">
+            <%= f.check_box :remember_me %>
+            <%= f.label :remember_me %>
+          </div>
```
Comment (OyePriscilla) [inline app/views/devise/sessions/new.html.erb:20]: - [ ] Please kindly fix this Log in text that overlapped other details for good user experience. ℹ️ 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @anyars-encarta, @faisalakb and @paulshaun-kdtk 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Install Ruby on Rails framework. ✔️ 
Use `params` from browser requests in a safe way. ✔️
Use preprocessed HTML file with embedded Ruby code. ✔️
Use layouts and templates for shared content. ✔️
Use database migration files to maintain database schema. ✔️
Set up associations between models. ✔️
Build a web app that requires the user to log in. ✔️
Use Devise gem for authentication. ✔️
Write unit tests using `Rspec` ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @faisalakb, @paulshaun-kdtk and @anyars-encarta  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application works as expected :white_check_mark: 
- [x] Application uses devise gem for user authentication :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by paulshaun-kdtk** [comment]
> # Approved 🟢
> Hi @faisalakb, @paulshaun-kdtk and @anyars-encarta
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉
> 
>  [  ](https://user-images.githubusercontent.com/38088818/253566971-43d45e44-98b5-4608-a589-53a38ffb3ea1.gif) [ ](https://user-images.githubusercontent.com/38088818/253566971-43d45e44-98b5-4608-a589-53a38ffb3ea1.gif)
> 
> ## Highlights ✔️
> * [x]  Good UI design ✅
> * [x]  Application works as expected ✅
> * [x]  Application uses devise gem for user authentication ✅
> * [x]  Git flow is used correctly ✅
> * [x]  Implemented all the changes from the previous reviewer ✅
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by paulshaun-kdtk** [comment]
@Felix45 and @OyePriscilla  Thank you for the constructive and helpful reviews

**Pull request merged**